### PR TITLE
Update dependencies and CAPI to 1.4

### PR DESCRIPTION
**Pull request by llDrLove**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
- Upgrade to Go 1.20.8
- Upgrade Kubernetes packages to 0.26.7
- Upgrade controller-runtime to v0.14.5
- Upgrade cluster-api to v1.4.5
- Upgrade CAPI to 1.4.5

**Special notes for your reviewer**:
NONE

**Documentation**:

NONE

**Release note**:

```release-note
Upgrade to Go 1.20.8 
Upgrade Kubernetes packages to 0.26.7
Upgrade controller-runtime to v0.14.5
Upgrade cluster-api to v1.4.5
Upgrade CAPI to 1.4.5
```

### CODE REVIEW

**Comment by cpanato** [comment]
/ok-to-test


I would like first we uograde to 0.14 make a release and then upgrade to 0.15

**Comment by llDrLove** [comment]
I've refactored the code to bump to controller-runtime `v0.14.5` and k8s dependencies `v0.26.7`. The CI e2e, lint and tests jobs are failing but it seems to be unrelated to my PR since the dependabot 🤖 [PR](https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/508), from last week, is also failing. Is it a known issue ?

**Comment by llDrLove** [comment]
@cpanato, could you review the PR ? I've addressed the CI pipeline failures I was having.

File: Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.19.10@sha256:83f9f840072d05ad4d90ce4ac7cb2427632d6b89d5ffc558f18f9577ec8188c0 as builder
+FROM golang:1.20.6@sha256:8e5a0067e6b387263a01d06b91ef1a983f90e9638564f6e25392fd2695f7ab6c as builder
```
Comment (cpanato) [inline Dockerfile:16]: i think for capi 1.4.x it is on go 1.19.x (https://github.com/kubernetes-sigs/cluster-api/blob/release-1.4/Makefile#L26C13-L26C22) we should match that similar to the other providers as well

File: scripts/go_install.sh [hunk 2]
```diff
@@ -40,6 +40,6 @@ fi
 rm "${GOBIN}/${2}"* || true
 
 # install the golang module specified as the first argument
-go install -tags tools "${1}@${3}"
+go install -tags capztools "${1}@${3}"
```
Comment (cpanato) [inline scripts/go_install.sh:43]: is this correct?

File: test/e2e/config/digitalocean-ci.yaml [hunk 3]
```diff
@@ -33,8 +33,8 @@ providers:
   - name: kubeadm
     type: ControlPlaneProvider
     versions:
-      - name: v1.3.8
-        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.8/control-plane-components.yaml
+      - name: v1.4.5
```
Comment (cpanato) [inline test/e2e/config/digitalocean-ci.yaml:36]: should change line 49 as well to `v1.4.99`

File: test/e2e/config/digitalocean-dev.yaml [hunk 4]
```diff
@@ -33,8 +33,8 @@ providers:
   - name: kubeadm
     type: ControlPlaneProvider
     versions:
-      - name: v1.3.8
-        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.8/control-plane-components.yaml
+      - name: v1.4.5
+        value: https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.5/control-plane-components.yaml
```
Comment (cpanato) [inline test/e2e/config/digitalocean-dev.yaml:37]: same here line 49

File: test/e2e/data/metadata/cluster-api/metadata.yaml [hunk 5]
```diff
@@ -1,6 +1,9 @@
 apiVersion: clusterctl.cluster.x-k8s.io/v1alpha3
 kind: Metadata
 releaseSeries:
+  - major: 1
```
Comment (cpanato) [inline test/e2e/data/metadata/cluster-api/metadata.yaml:4]: Need to do the same change in https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/blob/main/metadata.yaml

**Review by cpanato — CHANGES_REQUESTED** [review]
thank you for your changes!
have a few questions in some comments.

also can you squash the commits and update the PR summary and title? it is not just update dependencies it is upgrade CAPI to 1.4

File: Dockerfile (hunk 1 above)
Comment (llDrLove) [reply Dockerfile:16]: It seems like they have updated it to 1.20.8 in this commit for CAPI 1.4.x [here](https://github.com/kubernetes-sigs/cluster-api/commit/6b7d2b2680354340433606462f208ac934963939). I'll update to 1.20.8 if that is fine with you ?

File: test/e2e/data/metadata/cluster-api/metadata.yaml (hunk 5 above)
Comment (llDrLove) [reply test/e2e/data/metadata/cluster-api/metadata.yaml:4]: Sounds good! I've added it [here](https://github.com/kubernetes-sigs/cluster-api-provider-digitalocean/pull/509/files#diff-d186204ea7abb85bbadd9d94ab29d5bcefcc43d6a0ed06015e98a71b2f099423R22).

File: test/e2e/config/digitalocean-dev.yaml (hunk 4 above)
Comment (llDrLove) [reply test/e2e/config/digitalocean-dev.yaml:37]: Fixed

File: test/e2e/config/digitalocean-ci.yaml (hunk 3 above)
Comment (llDrLove) [reply test/e2e/config/digitalocean-ci.yaml:36]: Fixed

File: scripts/go_install.sh (hunk 2 above)
Comment (llDrLove) [reply scripts/go_install.sh]: I've reverted that change. I think it was an issue I had on macOS during debugging.

**Comment by llDrLove** [comment]
@cpanato , I've addressed your comment. Could you review again ?

File: Dockerfile (hunk 1 above)
Comment (cpanato) [reply Dockerfile]: sounds great!

**Comment by llDrLove** [comment]
@cpanato , did you have other comments regarding this PR ? Thanks !

**Comment by cpanato** [comment]
@llDrLove thanks for your work here, i think it needs a go mod update `go mod tidy` and a new push, can you do that?

**Comment by llDrLove** [comment]
@cpanato , yes I've fixed the issue with `go mod tidy`. CI is green ✅ .

**Review by cpanato — APPROVED** [review]
thank you
/approve
/lgtm


I will start the process to we make a release :D

cc @timoreimann @gottwald

**Pull request merged**